packages/core/src/mixins/as-text-input.scss

```
@@ -0,0 +1,44 @@
+@use '../helpers';
+@use './with-inner-focus' as mixins;
+
+@mixin as-text-input() {
+  @include helpers.font('300-regular');
+
+  $border-width: 1px;
+
+  background-color: helpers.color('background-input');
+  border: $border-width solid helpers.color('border-input');
+  border-radius: helpers.border-radius('medium');
+  box-sizing: border-box;
+  color: helpers.color('content-main');
+  padding: helpers.space(1.5) - $border-width helpers.space(2) - $border-width;
+  width: helpers.space(50);
+
+  &::placeholder {
+    color: helpers.color('content-placeholder');
+  }
+
+  &:hover {
+    border-color: helpers.color('border-input-hover');
+  }
+
+  // needs to be applied between :hover and .-invalid
```

Just to keep consistency with Radio component:

```suggestion
  // needs to be applied between [hover] and invalid states
```